In [1]:
# Which restaurant in Pune has the most number of people visiting?
# This problem can be solved by getting the number of reviews 
# Restaurant in pune has most number of peeople visiting
import numpy as np
import pandas as pd

data = pd.read_json('data.json')
# df = pd.DataFrame(data)

# data

In [28]:
# We'll access restaurants and do a function on each object and replace it
df = pd.DataFrame(data)
ratings_array = []
cnt = 0
ratingCountSet = set([])
def expenses(val):
    global cnt
    for rest in val['restaurants']: 
        try:
            ratingCountSet.add(val['restaurants'][rest]['rating_count'])
        except:
            continue
    # if(cnt == 0) :
        # print(val)

    cnt = cnt+1
    return val
    obj = val
    cost = 0
    cnt = 0
    if(type(obj) == float):
        # NaN value preprocessed
        return 0
    
    if('rating' in obj):
        lc_obj = {"name" : obj['name'], "rating": 0 , "address" : "Not in data"}
        try:
            lc_obj['rating'] = float(obj['rating'])
            cnt = cnt+1
            ratings_array.append(lc_obj)
            
        except:
            lc_obj = lc_obj
        
        try:
            lc_obj['address'] = obj['address']
        except:
            lc_obj = lc_obj
        
        return lc_obj
    else:
       return np.nan
        
    
# 11 problem 
df2 = df
df2 = df2['Pune']
# print(df2)
# # print(df2)
df2 = df2.dropna()
# # print(df2)
df2 = pd.DataFrame(df2)
# print(df2)
# print(df3)
df3 = pd.DataFrame(df2)
df3 = df3.drop('link',axis = 0)

df4 = df3['Pune'].apply(expenses)
# print(df3)
# df2 = df2['restaurants'].apply(expenses)

In [29]:
# list(ratingCountSet)

In [30]:
ratingCountSet.remove('NA')
ratingCountSet.remove('Too Few Ratings')
dict = {key : int(key.split()[0].replace('+','').replace('K','000')) for key in list(ratingCountSet)}
# dict

In [31]:
dict['NA'] = np.nan
dict['Too Few Ratings'] = 2
# dict

In [32]:
# We'll access restaurants and do a function on each object and replace it
df = pd.DataFrame(data)
noOfReviews = []
cnt = 0
def expenses(val):
    global cnt
    # val = dict[val]
    obj = val['restaurants']
    # if(cnt == 0):
                # print(obj)
    
    cnt = cnt+1
    # print(cnt)
    max_lc_obj = {}
    mx = 0
    for ele in obj:
        # if(cnt == 1) : 
            # print(ele)
        lc_obj = {"name" : obj[ele]['name'], "rating_count": 0 , "address" : "Not in data"}
        try:
            if(dict[obj[ele]['rating_count']] >= mx):
                 mx = dict[obj[ele]['rating_count']]
                 max_lc_obj = lc_obj
            lc_obj['rating_count'] = dict[obj[ele]['rating_count']]
            
            noOfReviews.append(lc_obj)
            
        except:
            lc_obj = lc_obj
        
        try:
            lc_obj['address'] = obj[ele]['address']
        except:
            lc_obj = lc_obj
        
    if(len(max_lc_obj) == 0) :
         return np.nan
    return max_lc_obj
        
# 11 problem 
# print(df3)
df4 = df3['Pune'].apply(expenses)


In [33]:

df5 = df4.dropna()
# df5

In [34]:
df5 = pd.DataFrame(df5)
df5.head()

Pune
Wanowrie         {'name': 'Ahad', 'rating_count': 2, 'address':...
Koregaon Park    {'name': 'THE CHICKSTER- Fried Chicken', 'rati...
Hadapsar         {'name': 'Hotel Garwa Satyampuram (Veg & NonVe...
Warje            {'name': 'Aroma's Hyderabad House(Warje)', 'ra...
Pimple Saudagar  {'name': 'Burger King', 'rating_count': 1000, ...

In [35]:
def deduction(val):
    # print(val)
    obj = val['rating_count']
    return obj
def deduction2(val):
    # print(val)
    obj = val['name']
    return obj
def deduction3(val):
    
    obj = val['address']
    return obj
df5['Address'] = df5['Pune'].apply(deduction3)
df5['Name'] = df5['Pune'].apply(deduction2)
df5['Pune'] = df5['Pune'].apply(deduction)
# df5.head()

In [36]:
df6 = df5.sort_values(by='Pune' , ascending = False)
df6.rename(columns = {'Pune':'Ratings'}, inplace = True)

df6[:10]

Ratings                                            Address   
Sinhgad Road      1000  Brahma Garden, Vishal Height, Manikbaug sinhga...  \
Nigdi             1000  Krishna Veg, 166, Dr Babasaheb Ambedkar Rd, Se...   
Baner             1000  Kalyan Bhel, Ganaraj chowk opp. Cosmos regency...   
Shivaji Nagar     1000  PK Biryani House (Deccan), Deccan Gymkhana, Pu...   
Kharadi           1000  Chaitanya Paranthas, Opposite Zensar IT Park, ...   
Kalyani Nagar     1000  McDonald's, Marigold Complex, Gr Floor, Survey...   
Aundh             1000  Anna Idli, Opposite Aman Honda Showroom, Baner...   
Viman Nagar       1000  Sandwich Express, Shop number 5 lunkad Plaza s...   
PCMC              1000  Burger King, Shop no 5/6, Foodcourt, Elprocity...   
Hinjewadi         1000  Desi Dhaba, NEAR LAXMI CHOWK, MEZZA9 ROAD CONN...   

                                    Name  
Sinhgad Road               Brahma Garden  
Nigdi                        Krishna Veg  
Baner                        Kalyan Bhel  
Shivaji Nagar  PK Biryani House (Deccan)  
Kharadi              Chaitanya Paranthas  
Kalyani Nagar                 McDonald's  
Aundh                          Anna Idli  
Viman Nagar             Sandwich Express  
PCMC                         Burger King  
Hinjewadi                     Desi Dhaba

Problem 11 : Which restaurant in Pune has the most number of people visiting?

	Brahma Garden, Vishal Height, Manikbaug sinhga...	Brahma Garden
    
    Krishna Veg, 166, Dr Babasaheb Ambedkar Rd, Se...	Krishna Veg	
    
    Kalyan Bhel, Ganaraj chowk opp. Cosmos regency...	Kalyan Bhel
    
	PK Biryani House (Deccan), Deccan Gymkhana, Pu...	PK Biryani House (Deccan)
    
    Chaitanya Paranthas, Opposite Zensar IT Park, ...	Chaitanya Paranthas
    
	McDonald's, Marigold Complex, Gr Floor, Survey...	McDonald's
    
    Aundh	1000	Anna Idli, Opposite Aman Honda Showroom, Baner...	Anna Idli
    
	Sandwich Express, Shop number 5 lunkad Plaza s...	Sandwich Express	
    
    Burger King, Shop no 5/6, Foodcourt, Elprocity...	Burger King	
    
    Desi Dhaba, NEAR LAXMI CHOWK, MEZZA9 ROAD CONN...	Desi Dhaba
    

In [37]:
print(df6[:5])

               Ratings                                            Address   
Sinhgad Road      1000  Brahma Garden, Vishal Height, Manikbaug sinhga...  \
Nigdi             1000  Krishna Veg, 166, Dr Babasaheb Ambedkar Rd, Se...   
Baner             1000  Kalyan Bhel, Ganaraj chowk opp. Cosmos regency...   
Shivaji Nagar     1000  PK Biryani House (Deccan), Deccan Gymkhana, Pu...   
Kharadi           1000  Chaitanya Paranthas, Opposite Zensar IT Park, ...   

                                    Name  
Sinhgad Road               Brahma Garden  
Nigdi                        Krishna Veg  
Baner                        Kalyan Bhel  
Shivaji Nagar  PK Biryani House (Deccan)  
Kharadi              Chaitanya Paranthas  


Problem  8 : Name the top 5 Most Popular Restaurants in Pune.


	Brahma Garden, Vishal Height, Manikbaug sinhga...	Brahma Garden
    
    Krishna Veg, 166, Dr Babasaheb Ambedkar Rd, Se...	Krishna Veg	
    
    Kalyan Bhel, Ganaraj chowk opp. Cosmos regency...	Kalyan Bhel
    
	PK Biryani House (Deccan), Deccan Gymkhana, Pu...	PK Biryani House (Deccan)
    
    Chaitanya Paranthas, Opposite Zensar IT Park, ...	Chaitanya Paranthas
    